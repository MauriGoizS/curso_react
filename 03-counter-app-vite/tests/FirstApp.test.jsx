const { render } = require("@testing-library/react")
const { FirstApp } = require("../src/FirstApp")

describe('Tests in <FirstApp />', () => {
    test('Must match the snapshot', () => {
        
        const title = 'Hi, I am Goku'
        const { container } = render( <FirstApp title={ title }/>);

        expect( container ).toMatchSnapshot();
    });
    
    test('Must display the title in a h1', () => {
        
        const title = 'Hi, I am Goku'
        const { container, getByText } = render( <FirstApp title={ title }/>);
        expect( getByText(title) ).toBeTruthy();

        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain( title );

    })
    

});
