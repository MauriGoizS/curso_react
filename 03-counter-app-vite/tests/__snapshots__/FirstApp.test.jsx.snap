// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp /> Must match the snapshot 1`] = `
<div>
  <h1>
    Hi, I am Goku
  </h1>
  <p>
    No hay subt√≠tulo
  </p>
  <p>
    Mauricio Goiz
  </p>
</div>
`;
